### PR TITLE
Multiple IP per nic - cmdlet changes

### PR DESCRIPTION
**Pull request by subashchandra31**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [ ] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [ ] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [ ] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.
- [ ] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

File: src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs [hunk 1]
```diff
@@ -237,33 +172,15 @@ public override void ExecuteSiteRecoveryCmdlet()
...
-            if (string.IsNullOrEmpty(this.RecoveryVMSubnetName) &&
-                !string.IsNullOrEmpty(this.RecoveryNicStaticIPAddress))
-            {
-                this.WriteWarning(Resources.RecoverySubnetInformationMissing);
-                return;
-            }
 
-            if (string.IsNullOrEmpty(this.TfoVMSubnetName) &&
-                !string.IsNullOrEmpty(this.TfoNicStaticIPAddress))
+            // how to verify if tfo/fo update input is provided and check corresponding network id.
```
Comment (punit1396) [inline src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs:176]: Check the condition, comment and the associated warning. The warning says recovery info missing but is being printed in case of a non-empty recovery network id.

File: src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.AzureToAzure,
+            Mandatory = true,
+            HelpMessage = "Specify the IP config name.")]
+        public string IpConfigName { get; set; }
+
+        /// <summary>
+        ///    Gets or sets the if the IP config is primary on NIC.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.AzureToAzure,
```
Comment (punit1396) [inline src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs:61]: switch parameter. This would have been flagged in the PS design review. Could we start the PS design review for these changes as well?

File: src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+            Mandatory = false,
+            HelpMessage = "Specify if the IP config is primary.")]
```
Comment (punit1396) [inline src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs:63]: Specifies whether*

File: src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+        public bool IsPrimary { get; set; }
+
+        /// <summary>
+        ///     Gets or sets whether an existing IP config is selected for tfo/failover.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.AzureToAzure,
+            Mandatory = false,
+            HelpMessage = "Specifies whether an existing IP config is selected for tfo/failover.")]
```
Comment (punit1396) [inline src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs:71]: test failover*

File: src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+        [ValidateNotNullOrEmpty]
```
Comment (punit1396) [inline src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs:72]: unnecessary

File: src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+
+        /// <summary>
+        ///     Gets or sets the target backend address pools for the test failover IP config.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.AzureToAzure,
+            Mandatory = false,
+            HelpMessage = "Specifies the IDs of backend address pools for the test failover IP config.")]
+        [ValidateNotNull]
+        public string[] TfoLBBackendAddressPoolIds { get; set; }
+
```
Comment (punit1396) [inline src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs:146]: nit. remove multiple blank lines

File: src/RecoveryServices/RecoveryServices.SiteRecovery/Properties/Resources.Designer.cs [hunk 7]
```diff
@@ -1036,7 +1047,18 @@ internal class Resources {
                 return ResourceManager.GetString("UnsupportedReplicationProvidedForASRVMNicConfig", resourceCulture);
             }
         }
-        
+
+        /// <summary>
+        ///   Looks up a localized string similar to Unsupported replication provider for ASR VM Nic Config..
+        /// </summary>
+        internal static string UnsupportedReplicationProvidedForASRVMNicIPConfig
```
Comment (punit1396) [inline src/RecoveryServices/RecoveryServices.SiteRecovery/Properties/Resources.Designer.cs:1054]: Corresponding resx file changes?

File: src/RecoveryServices/RecoveryServices.SiteRecovery/Properties/Resources.Designer.cs [hunk 8]
```diff
@@ -520,7 +520,18 @@ internal class Resources {
                 return ResourceManager.GetString("NicNotFoundInVM", resourceCulture);
             }
         }
-        
+
+        /// <summary>
+        ///   Looks up a localized string similar to IP config &quot;{0}&quot; not found in VM Nic..
+        /// </summary>
+        internal static string IPConfigNotFoundInVMNic
```
Comment (punit1396) [inline src/RecoveryServices/RecoveryServices.SiteRecovery/Properties/Resources.Designer.cs:527]: dont see the resx file changes for this new warning string

**Review by punit1396 — CHANGES_REQUESTED** [review]
Have left comments. Please start a cmdlet design PR as well.

File: src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs (hunk 1 above)
Comment (subashchandra31) [reply src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs:176]: existing logic had access to recovery/tfo subnet name.. hence can infer whether recovery/tfo network id is required and raise error if they are not provided.. but now  i moved them to ipconfigs..  so even if tfonetwork is not provided (if needed) given we are populating defaults from RPI in the NewAzureRmAsrVmNicIPConfig cmdlet, there is no info to check whether cx provided tfoNetworkId and its from the RPI.. i need to introduce a flag to track that from NewAzureRmAsrVmNicIPConfig  cmdlet (same for recovery network id). Correct me if i am missing something.

File: src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs (hunk 2 above)
Comment (subashchandra31) [reply src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs:61]: given we dont want to support changing ipconfig from primary to secondary and vice versa, is switch paramater the way to go !?

File: src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs (hunk 5 above)
Comment (subashchandra31) [reply src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs:72]: but this is the param based on which ipconfig is selected/not-selected for failover.. or u mean validatenotnullorempty is not required !?

File: src/RecoveryServices/RecoveryServices.SiteRecovery/Properties/Resources.Designer.cs (hunk 8 above)
Comment (subashchandra31) [reply src/RecoveryServices/RecoveryServices.SiteRecovery/Properties/Resources.Designer.cs]: added

File: src/RecoveryServices/RecoveryServices.SiteRecovery/Properties/Resources.Designer.cs (hunk 7 above)
Comment (subashchandra31) [reply src/RecoveryServices/RecoveryServices.SiteRecovery/Properties/Resources.Designer.cs:1053]: added

File: src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs (hunk 6 above)
Comment (subashchandra31) [reply src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs:146]: done

File: src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs (hunk 3 above)
Comment (subashchandra31) [reply src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs]: done

File: src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs (hunk 4 above)
Comment (subashchandra31) [reply src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs]: done

File: src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs (hunk 1 above)
Comment (punit1396) [reply src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs:176]: In that case, we need to add this validation to the IP config cmdlet and remove it from here

File: src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs (hunk 2 above)
Comment (punit1396) [reply src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs:61]: bools are not allowed as params through value. Instead should be through switch param. As for the need, that needs to be determined by us. But bool params aren't allowed like this

File: src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs (hunk 5 above)
Comment (punit1396) [reply src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs:72]: the Validate attribute is not required

File: src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs (hunk 1 above)
Comment (subashchandra31) [reply src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs]: moved validation and defaulting logic to NicConfig cmdlet as discussed to accommodate network validation.

File: src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs (hunk 5 above)
Comment (subashchandra31) [reply src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs]: done

File: src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs (hunk 2 above)
Comment (subashchandra31) [reply src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs]: removed this from input params.. used defaulting logic to set this info.

File: src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs [hunk 9]
```diff
@@ -40,7 +40,7 @@ public class NewAzureRmAsrVmNicConfig : SiteRecoveryCmdletBase
         public string NicId { get; set; }
 
         /// <summary>
-        ///    Gets or sets the NIC Id.
+        ///    Specify the ASR Replication Protected Item.
```
Comment (punit1396) [inline src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs:43]: get or sets *

File: src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs [hunk 10]
```diff
@@ -294,23 +203,19 @@ public override void ExecuteSiteRecoveryCmdlet()
...
-                        this.RecoveryVMNetworkId =
-                            this.ReplicationProtectedItem.SelectedRecoveryAzureNetworkId;
-                    }
-
-                    if (!this.MyInvocation.BoundParameters.ContainsKey(
-                            Utilities.GetMemberName(() => this.RecoveryVMSubnetName)))
+                    if (this.IPConfigs != null)
                     {
-                        this.RecoveryVMSubnetName = vmNic.RecoveryVMSubnetName;
+                        if (!validateAndPopulateIPConfigs(vmNic))
```
Comment (punit1396) [inline src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs:208]: method name - Pascal case instead of camel

Comment (subashchandra31) [reply src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs]: updated

File: src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs (hunk 9 above)
Comment (subashchandra31) [reply src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs]: done

File: src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs [hunk 11]
```diff
@@ -374,69 +253,115 @@ public override void ExecuteSiteRecoveryCmdlet()
...
 
                     break;
             }
 
             this.WriteObject(nicConfig);
         }
+
+        // validates IP configs and populate values not provided in input from DB NIC entiry.
+        // returns false in case of validation error.
+        private bool validateAndPopulateIPConfigs(ASRVMNicDetails vmNic)
```
Comment (punit1396) [inline src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs]: Pascal case

Comment (subashchandra31) [reply src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs]: updated.. in above commit.

File: src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs [hunk 12]
```diff
@@ -374,69 +253,115 @@ public override void ExecuteSiteRecoveryCmdlet()
...
+                {
+                    ipConfig.RecoverySubnetName = vmNicIPConfig.RecoverySubnetName;
+                }
+
+                if (ipConfig.RecoveryStaticIPAddress == null)
+                {
+                    ipConfig.RecoveryStaticIPAddress = vmNicIPConfig.RecoveryStaticIPAddress;
+                }
+
+                if (string.IsNullOrEmpty(ipConfig.RecoveryPublicIPAddressId))
```
Comment (punit1396) [inline src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs:321]: only null check. Update the ValidateIsNotNullOrEmpty to just ValidateIsNotNull

File: src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs [hunk 13]
```diff
@@ -374,69 +253,115 @@ public override void ExecuteSiteRecoveryCmdlet()
...
+                {
+                    ipConfig.RecoveryPublicIPAddressId = vmNicIPConfig.RecoveryPublicIPAddressId;
+                }
+
+                if (ipConfig.RecoveryLBBackendAddressPoolIds.Count == 0)
```
Comment (punit1396) [inline src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs:326]: only null check. Allow reset

File: src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs [hunk 14]
```diff
@@ -374,69 +253,115 @@ public override void ExecuteSiteRecoveryCmdlet()
...
+                {
+                    ipConfig.TfoSubnetName = vmNicIPConfig.TfoSubnetName;
+                }
+
+                if (ipConfig.TfoStaticIPAddress == null)
+                {
+                    ipConfig.TfoStaticIPAddress = vmNicIPConfig.TfoStaticIPAddress;
+                }
+
+                if (string.IsNullOrEmpty(ipConfig.TfoPublicIPAddressId))
```
Comment (punit1396) [inline src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs:341]: same for tfo public IP and LB

File: src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs [hunk 15]
```diff
@@ -0,0 +1,171 @@
...
+        [ValidateNotNull]
+        public string RecoveryStaticIPAddress { get; set; }
+
+        /// <summary>
+        ///     Gets or sets the id of the public IP address resource associated with the recovery IP config.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.AzureToAzure,
+            Mandatory = false,
+            HelpMessage = "Specifies the ID of the public IP address associated with the recovery IP config.")]
+        [ValidateNotNullOrEmpty]
```
Comment (punit1396) [inline src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs:73]: as mentioned update this and tfo public ip

File: src/RecoveryServices/RecoveryServices.SiteRecovery/Properties/Resources.resx [hunk 16]
```diff
@@ -480,6 +480,9 @@ Recommended Action: To cleanup the resources created by Test failover run the Te
   <data name="NicNotFoundInVM" xml:space="preserve">
     <value>Nic "{0}" not found in VM.</value>
   </data>
+  <data name="IPConfigNotFoundInVMNic" xml:space="preserve">
+    <value>IP Config "{0}" not found in VM NIC.</value>
```
Comment (punit1396) [inline src/RecoveryServices/RecoveryServices.SiteRecovery/Properties/Resources.resx:484]: considering multiple NICs, let's add nic info in the output as well. preferably name

Comment (punit1396) [inline src/RecoveryServices/RecoveryServices.SiteRecovery/Properties/Resources.resx:484]: why is this added in the middle. Add at the end please

File: src/RecoveryServices/RecoveryServices.SiteRecovery/ReplicationProtectedItem/SetAzureRmRecoveryServicesAsrReplicationProtectedItem.cs [hunk 17]
```diff
@@ -820,18 +812,22 @@ private List<VMNicInputDetails> getNicListToUpdate(IList<VMNicDetails> vmNicList
                     else
                     {
                         vMNicInputDetails.NicId = nDetails.NicId;
-                        vMNicInputDetails.RecoveryVMSubnetName = nDetails.RecoveryVMSubnetName;
-                        vMNicInputDetails.ReplicaNicStaticIPAddress =
-                            nDetails.ReplicaNicStaticIPAddress;
                         vMNicInputDetails.SelectionType = nDetails.SelectionType;
-                        vMNicInputDetailsList.Add(vMNicInputDetails);
+                        vMNicInputDetails.IpConfigs = nDetails.IpConfigs?.Select(ip =>
```
Comment (punit1396) [inline src/RecoveryServices/RecoveryServices.SiteRecovery/ReplicationProtectedItem/SetAzureRmRecoveryServicesAsrReplicationProtectedItem.cs:816]: Null check - ?.Select()?.ToList()

File: src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs (hunk 12 above)
Comment (subashchandra31) [reply src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs]: updated

File: src/RecoveryServices/RecoveryServices.SiteRecovery/ReplicationProtectedItem/SetAzureRmRecoveryServicesAsrReplicationProtectedItem.cs (hunk 17 above)
Comment (subashchandra31) [reply src/RecoveryServices/RecoveryServices.SiteRecovery/ReplicationProtectedItem/SetAzureRmRecoveryServicesAsrReplicationProtectedItem.cs:816]: done

File: src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs (hunk 13 above)
Comment (subashchandra31) [reply src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs]: done

File: src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs (hunk 14 above)
Comment (subashchandra31) [reply src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs]: done

File: src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs (hunk 15 above)
Comment (subashchandra31) [reply src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs]: done

File: src/RecoveryServices/RecoveryServices.SiteRecovery/Properties/Resources.resx (hunk 16 above)
Comment (subashchandra31) [reply src/RecoveryServices/RecoveryServices.SiteRecovery/Properties/Resources.resx]: updated and moved to end

Comment (subashchandra31) [reply src/RecoveryServices/RecoveryServices.SiteRecovery/Properties/Resources.resx]: updated and moved to end

**Review by punit1396 — APPROVED** [review]
Changes look fine. However, do verify and test these out.
